buildSrc/src/main/kotlin/Documentation.kt

```diff
@@ -10,22 +10,23 @@
 
 import org.gradle.api.Project
 import org.gradle.api.tasks.TaskProvider
-import org.gradle.jvm.tasks.Jar
+import org.gradle.api.tasks.bundling.Jar
 import org.jetbrains.dokka.gradle.DokkaTask
 import org.gradle.kotlin.dsl.*
 
-fun Project.createJavadocJar(): TaskProvider<Jar> {
+fun Project.createJavadocJar(mainSourceSetPath: String = "src/main"): TaskProvider<Jar> {
     val dokkaHtml = tasks.named<DokkaTask>("dokkaHtml") {
         outputDirectory.set(file("${layout.buildDirectory.get()}/javadoc"))
         dokkaSourceSets {
             named("main") {
-                sourceRoots.from(file("src/jvm/main"))
+                sourceRoots.from(file(mainSourceSetPath))
                 reportUndocumented.set(false)
             }
         }
     }
 
     val javadocJar = tasks.register<Jar>("javadocJar") {
+        archiveBaseName.set(this@createJavadocJar.name)
```

Comment:
Are you sure this is required? Before this change, the name of the generated javadoc was e.g., `lincheck-trace-javadoc.jar` for the `trace` artifact, which was expected behavior I think. What would be the name after this change?